### PR TITLE
[FEATURE] Extend configuration

### PR DESCRIPTION
**Pull request by DominikBucher12**
# What this PR do:
- Extends Configuration to contain things like autocorrection type and Spell checking.
# How to test it
- Run the app, tests etc. Try different configurations

### CODE REVIEW

File: Tests/RichTextKitTests/RichTextViewRepresentableTests.swift [hunk 1]
```diff
@@ -45,9 +45,10 @@ final class RichTextViewComponentTests: XCTestCase {
     func testSettingUpWithEmptyTextWorks() {
         let string = NSAttributedString(string: "")
         view.setup(with: string, format: .rtf)
+        view.configuration = .standard
```
Comment (DominikBucher12) [inline Tests/RichTextKitTests/RichTextViewRepresentableTests.swift:48]: This raises question if configuration shouldn't be passed to setup instead... This feels like API is all over the place, your thoughts @danielsaidi ?

**Comment by DominikBucher12** [comment]
Btw this fixes https://github.com/danielsaidi/RichTextKit/issues/132 😍

File: Tests/RichTextKitTests/RichTextViewRepresentableTests.swift (hunk 1 above)
Comment (danielsaidi) [reply Tests/RichTextKitTests/RichTextViewRepresentableTests.swift:48]: I'm thinking we should inject it into the environment in SwiftUI.

**Review by danielsaidi — APPROVED** [review]

**Comment by danielsaidi** [comment]
Great job!

**Pull request merged**